docs/documentation/siddhi-4.0.md

```diff
@@ -1754,6 +1754,176 @@ define stream TempStream (deviceID long, roomNo int, temp double);
+
+In order to execute store queries, there should be a siddhi app deployed with a store defined, that contains the table which is going to be queried.
+
+**Example**
+
+If user needs to query the table named "**RoomTypeTable**", the it should have been defined in the siddhi app.
+
+In order to execute a store query on **RoomTypeTable**, user needs to submit the store query using query() method of SiddhiAppRuntime intance as below.
+
+```java
+siddhiAppRuntime.query(<store query>);
+```
+
+### Select
+
+SELECT store query supports to retrieve one or more records that match with a given condition, from a given table.
+
+**Syntax**
+
+```sql
+from <table>
+select <attribute name>, <attribute name>, ...
+group_by? having? order_by? limit?
+```
+
+**Example**
+
+Following example shows how to retrieve room numbers and types of the rooms starting from the room no 10.
+
+```sql
+from roomTypeTable
+select roomNo, type
+on roomNo >= 10;
+```
+
+### Delete
+
+DELETE store query can be used to delete selected records from a given database table.
+
+**Syntax**
+
+```sql
+<select>?  
+delete <table>  on <conditional_expresssion>
+```
+
+The `condition` element specifies the basis on which records are selected to be deleted.
+
+!!! note
+    When specifying the condition,table attributes must be always referred to with the table name as, 
+    `<table name>.<attibute name>`
+
+**Example**
+
+In this example, query deletes a record in the table named `RoomTypeTable` if it has value for the `roomNo` 
+attribute that matches the value for the `roomNumber` attribute of the selection which has 10 as the actual value.
+
+```sql
+select 10 as roomNumber
+delete RoomTypeTable
+on RoomTypeTable.roomNo == roomNumber;
+```
+
+### Update
+
+Update store query can be used to update selected attributes stored in a table based on a condition.
+
+**Syntax**
+
+```sql
+select <attribute name>, <attribute name>, ...?
+update <table>
+    set <table>.<attribute name> = (<attribute name>|<expression>)?, <table>.<attribute name> = (<attribute name>|<expression>)?, ...
+    on <condition>
+```
+
+The `condition` element specifies the basis on which records are selected to be updated.
+When specifying the `condition`, table attributes must be referred to with the table name.
+
+You can use the `set` keyword to update selected attributes from the table. Here, for each assignment, the attribute specified in the left must be the table attribute, and the one specified in the right can be a stream/table attribute a mathematical operation, or other. When the `set` clause is not provided, all the attributes in the table are updated.
+
+
+!!! note
+    Table attributes must be always referred to with the table name as, 
+     `<table name>.<attibute name>`.
+
+**Example**
+
+The following query updates the room occupancy by increasing its value by 1, in the `RoomOccupancyTable` table for each room number greater than 10.
```

Comment:
**Suggestion**: "This query updates the room occupancy by increasing the value for the `roomNo` column of the `RoomTypeTable` table by 1, for each record with a value for the `roomNo` column that is greater than `10`.